pathutil/pathutil.go

```diff
@@ -10,64 +10,54 @@ import (
 	"strings"
 )
 
-// TempDirProvider ...
-type TempDirProvider interface {
-	CreateTempDir(prefix string) (string, error)
+//
+// Path provider functions
+
+// PathProvider ...
+type PathProvider interface {
+	TempDir(prefix string) (string, error)
```

Comment:
If this creates a directory, shouldn't we keep the `Create` prefix in the function name?